app/models/spree/virtual_gift_card.rb

```diff
@@ -31,8 +36,13 @@ def redeem(redeemer)
     self.update_attributes( redeemed_at: Time.now, redeemer: redeemer )
   end
 
-  def make_redeemable!(purchaser:)
-    update_attributes!(redeemable: true, purchaser: purchaser, redemption_code: (self.redemption_code || generate_unique_redemption_code))
+  def make_redeemable!(purchaser:, inventory_unit:)
+    update_attributes!(redeemable: true, purchaser: purchaser, inventory_unit: inventory_unit, redemption_code: (self.redemption_code || generate_unique_redemption_code))
+  end
+
+  def deactivate!
+    cancel_and_reimburse_inventory_unit
+    update_attributes!(redeemable: false, deactivated_at: Time.now)
```

Comment:
i wonder, does it make sense to cancel the gift card before giving them back their money? if something fails on the deactivation then they would get their money back and still be able to redeem the card